WordPress/src/main/java/org/wordpress/android/ui/prefs/WPIconWidgetPreference.java

```diff
@@ -55,9 +58,15 @@ protected void onBindView(@NonNull View view) {
             button.setOnClickListener(new View.OnClickListener() {
                 @Override
                 public void onClick(View v) {
-                    getOnPreferenceClickListener().onPreferenceClick(wpButtonPreference);
+                    getOnPreferenceClickListener().onPreferenceClick(iconWidgetPreference);
                 }
             });
         }
+
+        if (view.findViewById(R.id.domain) != null) {
```

Comment:
Good point. I've changed it to be WPDeleteSitePreference. Once it can be used for a more generic use-case is when we'll figure out how to generify it.